lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -1430,7 +1430,7 @@ sub update_status ($self, $status) {
     #       wins the race updating the job state.
     $self->discard_changes;
     my $state = $self->state;
-    return {result => 0} unless $state eq RUNNING || $state eq UPLOADING;
+    return {result => 0} unless $state eq RUNNING || $state eq UPLOADING || $state eq CANCELLED;
```

Comment:
this feels like we want to combine the three individual states into a named condition

Reply:
Supposedly
```suggestion
    return {result => 0} if $state eq DONE;
```
would work as well. Of course this then includes also scheduled and assigned and one definitely had to test whether this actually works but I guess it wouldn't be a problem.